apps/tasks.py

```diff
@@ -649,3 +503,158 @@ def delete_old_objects():
     old_apps = AppInstance.objects.filter(created_on__lt=threshold_time, app__category__name="Develop")
     for app_ in old_apps:
         delete_resource.delay(app_.pk)
+
+
+@shared_task(bind=True, max_retries=None)
+def init_event_listener(self, namespace, label_selector):
+    """
+    The event listener takes the latest event and checks if a corresponding appinstance
+    should be updated. It uses the creation timestamp to always use the status of the youngest pod
+    in a helm release.
+    """
+    api, w = setup_client()
+    try:
+        my_dict = {}
+        for event in w.stream(api.list_namespaced_pod, namespace=namespace, label_selector=label_selector):
+            pod = event["object"]
+
+            status = get_status(pod)
+            status = status[:15]
+            release = pod.metadata.labels["release"]
+            creation_timestamp = pod.metadata.creation_timestamp
+            deletion_timestamp = pod.metadata.deletion_timestamp
+
+            appinstance = AppInstance.objects.filter(parameters__contains={"release": release}).last()
+
+            if appinstance:
+                # Case 1 - Set unseen release
+                if release not in my_dict:
+                    my_dict[release] = {
+                        "creation_timestamp": creation_timestamp,
+                        "deletion_timestamp": deletion_timestamp,
+                        "status": status,
+                    }
+
+                # If older pod, skip
+                if creation_timestamp < my_dict[release]["creation_timestamp"] or status == my_dict[release]["status"]:
+                    continue
+
+                # If pod is same and deleted, set deleted stamp
+                elif creation_timestamp == my_dict[release]["creation_timestamp"] and deletion_timestamp:
+                    status = "Deleted"
+                    appinstance.deleted_on = timezone.now()
+
+                # If pod is newer, update
+                my_dict[release] = {
+                    "creation_timestamp": creation_timestamp,
+                    "deletion_timestamp": deletion_timestamp,
+                    "status": status,
+                }
+                status_object = AppStatus(appinstance=appinstance)
+                update_status(appinstance, status_object, status)
+    except Exception as exc:
```

Comment:
Do you know explicit exceptions that could be caught?

Reply:
> Viktor Sandström
  [7 hours ago](https://scilifelab.slack.com/archives/D04AB9HQW8H/p1701419643671769?thread_ts=1701418543.336129&cid=D04AB9HQW8H)
Here is the exception that i've gotten a few times. Do not know what it means or why it shows up
celery.utils.serialization.UnpickleableExceptionWrapper: ProtocolError("Connection broken: InvalidChunkLength(got length b'', 0 bytes read)", InvalidChunkLength(got length b'', 0 bytes read))
Should i catch this?

I think it would be nice to at least print `exc`, so that we could trace it in case it would cause any issues